### DISCUSSION TITLE
Why this special form of identifier? Why not just `reachedScanCode`?

### DISCUSSION THREAD

**Discussion by vsajip** [congo-cc/congo-parser-generator#135] [Q&A] [answered]
https://github.com/congo-cc/congo-parser-generator/blob/430e15d542bbde75750b951a0a1366c8663161d9/src/templates/java/LookaheadRoutines.java.ftl#L132

This might be some Java feature that just passed me by ... looking to learn.

**Comment by revusky** [comment]
No reason really. I was just toying with the idea that variables in generated code should start/end with a `$`. But it's not applied generally, as you see.

**Comment by vsajip** [comment]
OK - IMO it hurts readability a bit. Not a problem for most users, but for us when troubleshooting code generation, and stepping through code generated when needed.

**Reply by revusky** [reply, marked as answer]
Well, like I say, I was toying with the idea of having all these variables follow some naming pattern that an application programmer would never use. But really, it's probably an extremely rare problem anyway. If you reallly hate it, you can change it back even. I don't really care very much.
